feat: case card drag within level (PT-188001450)

emcelroy opened the pull request:
[#188001450](https://www.pivotaltracker.com/story/show/188001450)

kswenson reviewed (approved):
👍 Looks good -- I added a commit which refactored the case-table/case-card dependencies from common code. Take a look, give it a whirl, and make sure I didn't break anything.

emcelroy commented:
> 👍 Looks good -- I added a commit which refactored the case-table/case-card dependencies from common code. Take a look, give it a whirl, and make sure I didn't break anything.

@kswenson Thanks 👍 Changes look good and everything seems to work the same in the browser.

@tealefristoe happened to notice an errant call to `createCasesNotification` in the `handleSubmit` function of `CaseAttrView`. Since that function only adds/edits attributes, not cases, I removed the call in the latest commit.

kswenson reviewed (approved):
👍 Changes LGTM
